### PR TITLE
Allow multiple null data outputs

### PR DESCRIPTION
**Pull request by jcramer**
This pr gives the minor updates required to allow multiple op_return outputs into the mempool with the same maximum null data allowance.

Tests modified or added:

1. Existing test `"More than one nulldata output"` is updated to pass as a standard transaction.
2. New test `"Multiple null data outputs exceeds MaxDataCarrierSize (non-standard)"` is added to show `MaxDataCarrierSize` is exceeded when multiple null data outputs are used with a first null data output having 223 bytes and a second null data output having 1 byte (i.e., 223 + 1 > 223).
3. New test `"Multiple null data outputs are below MaxDataCarrierSize (standard)"` is added to show `MaxDataCarrierSize` is not exceeded when multiple null data outputs are used with a first null data output having 222 bytes and a second null data output having 1 byte (i.e., 222 + 1 = 223).

### CODE REVIEW

**Comment by jcramer** [comment]
I replaced the hard-coded `223` in the unit tests with `txscript.MaxDataCarrierSize`.

**Comment by jcramer** [comment]
Renamed `numNullDataSize` to `dataCarrierSize` for consistency..

**Review by zquestz — APPROVED** [review]

File: mempool/policy.go [hunk 1]
```diff
@@ -321,8 +321,8 @@ func checkTransactionStandard(tx *bchutil.Tx, height int32,
 
 	// A standard transaction must not have more than one output script that
 	// only carries data.
-	if numNullDataOutputs > 1 {
-		str := "more than one transaction output in a nulldata script"
+	if dataCarrierSize > maxDataCarrierSize {
+		str := "transaction nulldata exceeds 223 bytes"
```
Comment (zquestz) [inline mempool/policy.go:325]: Shouldn't this use maxDataCarrierSize? That way the message changes when we update the size later. ;)

File: mempool/policy_test.go [hunk 2]
```diff
@@ -530,13 +530,115 @@ func TestCheckTransactionStandard(t *testing.T) {
...
+			code:       wire.RejectNonstandard,
+		},
 	}
 
 	pastMedianTime := time.Now()
 	for _, test := range tests {
 		// Ensure standardness is as expected.
 		err := checkTransactionStandard(bchutil.NewTx(&test.tx),
-			test.height, pastMedianTime, DefaultMinRelayTxFee, 1)
+			test.height, pastMedianTime, DefaultMinRelayTxFee, 1, 223)
```
Comment (zquestz) [inline mempool/policy_test.go:641]: Should this test use the constant `maxDataCarrierSize`? Maybe we can export it. =)

File: mempool/policy.go (hunk 1 above)
Comment (zquestz) [reply mempool/policy.go:325]: Or just use `txscript.MaxDataCarrierSize`

File: mempool/policy_test.go (hunk 2 above)
Comment (jcramer) [reply mempool/policy_test.go]: updated.

File: mempool/policy.go (hunk 1 above)
Comment (jcramer) [reply mempool/policy.go:325]: updated.

File: mempool/mempool_test.go [hunk 3]
```diff
@@ -6,13 +6,14 @@ package mempool
...
 	"encoding/hex"
-	"github.com/gcash/bchd/mining"
 	"reflect"
 	"runtime"
 	"sync"
 	"testing"
 	"time"
 
+	"github.com/gcash/bchd/mining"
+
```
Comment (zquestz) [inline mempool/mempool_test.go:16]: NIT: No need for newline.

File: mempool/policy.go (hunk 1 above)
Comment (zquestz) [reply mempool/policy.go:325]: I meant in the str line!

File: mempool/mempool_test.go (hunk 3 above)
Comment (jcramer) [reply mempool/mempool_test.go]: fixed

File: mempool/policy.go (hunk 1 above)
Comment (jcramer) [reply mempool/policy.go]: Got it, this is fixed now too.

**Review by zquestz — APPROVED** [review]

**Pull request merged**